src/test/groovy/io/openliberty/tools/gradle/DevTest.groovy

```diff
@@ -222,31 +218,36 @@ class DevTest extends BaseDevTest {
         File helloBatchObj = new File(targetDir, "classes/com/demo/HelloBatch.class");
         verifyFileExists(helloBatchObj, 15000);
         // ... and run the proper task.
-        assertTrue(verifyLogMessage(10000, RUNNING_GENERATE_FEATURES, ++generateFeaturesCount));
+        assertTrue(verifyLogMessage(10000, RUNNING_GENERATE_FEATURES, ++runGenerateFeaturesCount));
         assertTrue(verifyFileExists(newFeatureFile, 5000)); // task created file
         assertTrue(verifyFileExists(newTargetFeatureFile, 5000)); // dev mode copied file
         assertTrue(verifyLogMessage(10000, "batch-1.0", newFeatureFile));
         assertTrue(verifyLogMessage(10000, NEW_FILE_INFO_MESSAGE, newFeatureFile));
         assertTrue(verifyLogMessage(10000, SERVER_XML_COMMENT, serverXmlFile));
         // should appear as part of the message "CWWKF0012I: The server installed the following features:"
-        assertTrue(verifyLogMessage(10000, "batch-1.0", errFile));
+        assertTrue(verifyLogMessage(123000, SERVER_INSTALLED_FEATURES, errFile, ++installedFeaturesCount));
 
         // When there is a compilation error the generate features process should not run
         final String goodCode = "import javax.ws.rs.GET;";
         final String badCode  = "import javax.ws.rs.GET";
-        final String errMsg = "Source compilation had errors.";
-        int errCount = countOccurrences(errMsg, logFile);
+        int errCount = countOccurrences(COMPILATION_ERRORS, logFile);
         replaceString(goodCode, badCode, helloBatchSrc);
 
-        assertTrue(verifyLogMessage(10000, errMsg, errCount+1)); // wait for compilation
+        assertTrue(verifyLogMessage(10000, COMPILATION_ERRORS, errCount+1)); // wait for compilation
         int updatedgenFeaturesCount = countOccurrences(RUNNING_GENERATE_FEATURES, logFile);
-        assertEquals(generateFeaturesCount, updatedgenFeaturesCount);
+        // after failed compilation generate features is not run.
+        assertEquals(runGenerateFeaturesCount, updatedgenFeaturesCount);
 
         // Need valid code for testing
-        String goodCompile = "Source compilation was successful.";
-        int goodCount = countOccurrences(goodCompile, logFile);
+        int goodCount = countOccurrences(COMPILATION_SUCCESSFUL, logFile);
+        int regenerateCount = countOccurrences(REGENERATE_FEATURES, logFile);
         replaceString(badCode, goodCode, helloBatchSrc);
-        assertTrue(verifyLogMessage(10000, goodCompile, goodCount+1));
+        assertTrue(verifyLogMessage(10000, COMPILATION_SUCCESSFUL, goodCount+1));
+        // after successful compilation run generate features. "Regenerate" message should appear after.
+        // Restore these tests once issue 757 is fixed.
```

Comment:
Suggestion to mark as TODO so that it shows up in problems views of IDEs

Reply:
Done and moved to file DevRecompileTest.java